backend/modules/chats/agent.py

```diff
@@ -0,0 +1,236 @@
+from dataclasses import dataclass
+from datetime import date, datetime, timedelta
+from decimal import Decimal
+
+from asgiref.sync import sync_to_async
+from pydantic_ai import Agent, RunContext
+
+from config.clerk import client as clerk_client
+from modules.chats.mcps import MassiveMCP
+from modules.investors.models import Asset, Investor
+from modules.prices.repositories import PolygonPricesRepository
+from modules.transactions.models import Transaction
+
+massive_mcp = MassiveMCP()
+
+
+@dataclass
+class AgentDeps:
+    investor_id: str
+
+
+agent = Agent[AgentDeps, str](
+    deps_type=AgentDeps,
+    system_prompt=(
+        "You are a financial assistant for InvestLab, a paper trading simulator."
+    ),
+    toolsets=[
+        massive_mcp.get_toolset(),
+    ],
+)
+
+
+@agent.system_prompt
+async def add_the_users_first_name(ctx: RunContext[AgentDeps]) -> str:
+    investor = await sync_to_async(Investor.objects.get)(id=ctx.deps.investor_id)
+    user = await sync_to_async(clerk_client.users.get)(user_id=investor.clerk_id)
+    if not user:
+        return "The user's name is unknown."
+    return f"The user's name is {user.first_name}."
+
+
+@agent.system_prompt
+def add_the_date() -> str:
+    return f"The date is {date.today()}."
+
+
+# TODO: Refactor tool functions
+
+
+@agent.tool
+async def get_portfolio(ctx: RunContext[AgentDeps]) -> dict:
+    investor_id = ctx.deps.investor_id
+    investor = await sync_to_async(Investor.objects.get)(id=investor_id)

```

Comment:
a function that does the exact same thing exists in the statistics module - perhaps reuse it here? if the calculations are done in the view itself, just extract them to a function and reuse. This way we will get a guarantee that the chat will output the same thing as the portfolio summary widget